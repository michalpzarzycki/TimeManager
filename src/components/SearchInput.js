import React, { useState, useEffect } from 'react';
import styles from './SearchInput.module.css'
import { db } from '../firebase/firebase';

export default function SearchInput({searchValueSetter}) {
    let [searchResult,setSearchResult] = useState([])

     async function searchFunction(search) {
         let result=""
        console.log("PRZD SNAP", search)
        const snapshot = await db.collection('tasks')
                                .where('keyword', 'array-contains', search.toLowerCase())
                                .get().then(x => {
                                  result = x.forEach(x => setSearchResult([...searchResult, x.data()]))
                                })
                return result            
            
    }
    function handleChange(event) {
        searchValueSetter(event.target.value)
    //     db.collection('tasks').where('keyword', 'array-contains', event.target.value).get().then(snapshot => {
    //         snapshot.docs.forEach(doc => {
    //             console.log("DONE", doc.id, doc.data())
    //         })
          
    // })
}
    return (<>
        <div className={styles.route1TasksNav_searchDiv}>
        <span className={styles.span}>
        <input onChange={handleChange} name="search" type="search" placeholder="Search Task" autoComplete="off" className={styles.route1TasksNav_searchDiv_input}/>
        </span>
    </div>
    </>
    )
}