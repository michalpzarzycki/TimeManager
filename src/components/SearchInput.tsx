import React, { useState, useEffect } from 'react';
import styles from './SearchInput.module.css';
import { connect } from 'react-redux'
import { db } from '../firebase/firebase';
import { setSearchFilterField } from '../redux/action';

function SearchInput({searchFilterField ,onSearchChange} : any) {
    useEffect(() => {
        console.log("STATE", searchFilterField)
    },[searchFilterField])
    //  async function searchFunction(search) {
    //      let result=""
    //     console.log("PRZD SNAP", search)
    //     const snapshot = await db.collection('tasks')
    //                             .where('keyword', 'array-contains', search.toLowerCase())
    //                             .get().then(x => {
    //                               result = x.forEach(x => setSearchResult([...searchResult, x.data()]))
    //                             })
    //             return result            
            
    // }
//     function handleChange(event) {
//         searchValueSetter(event.target.value)
//     //     db.collection('tasks').where('keyword', 'array-contains', event.target.value).get().then(snapshot => {
//     //         snapshot.docs.forEach(doc => {
//     //             console.log("DONE", doc.id, doc.data())
//     //         })
          
//     // })
// }
    return (<>
        <div className={styles.route1TasksNav_searchDiv}>
        <span className={styles.span}>
        <input onChange={onSearchChange} name="search" type="search" placeholder="Search Task" autoComplete="off" className={styles.route1TasksNav_searchDiv_input}/>
        </span>
    </div>
    </>
    )
}

const mapStateToProps = (state : any) => {
    return {
        searchFilterField: state.searchFilterField
    }
}
const mapDisatchToProps = (dispatch : any) => {
    return {
        onSearchChange: (event : any) => dispatch(setSearchFilterField(event.target.value))
    }
}
export default connect(mapStateToProps, mapDisatchToProps)(SearchInput)