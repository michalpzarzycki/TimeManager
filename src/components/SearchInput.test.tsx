import React from 'react'
import {shallow} from 'enzyme'
import SearchInput from './SearchInput'
import {Provider} from 'react-redux'
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});

test('Expext to render SearchInput component', () => {
    expect(shallow(<Provider store={store}><SearchInput/></Provider>)).toMatchSnapshot()
})