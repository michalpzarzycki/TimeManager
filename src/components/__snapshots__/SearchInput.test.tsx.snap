// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expext to render SearchInput component 1`] = `ShallowWrapper {}`;
